src/resolve/resolved.c

```diff
@@ -46,35 +42,20 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
-        umask(0022);
-
-        r = mac_selinux_init();
-        if (r < 0) {
-                log_error_errno(r, "SELinux setup failed: %m");
-                goto finish;
-        }
-
-        r = get_user_creds(&user, &uid, &gid, NULL, NULL);
-        if (r < 0) {
-                log_error_errno(r, "Cannot resolve user name %s: %m", user);
+        if (geteuid() == 0) {
+                log_error("This program must be run under an unprivileged user"
+                          " with the following capabilities: CAP_NET_RAW"
+                          " CAP_NET_BIND_SERVICE CAP_SETPCAP");
+                r = -EINVAL;
                 goto finish;
         }
 
-        /* Always create the directory where resolv.conf will live */
-        r = mkdir_safe_label("/run/systemd/resolve", 0755, uid, gid);
+        r = mac_selinux_init();
         if (r < 0) {
-                log_error_errno(r, "Could not create runtime directory: %m");
+                log_error_errno(r, "SELinux setup failed: %m");
                 goto finish;
         }
 
-        /* Drop privileges, but keep three caps. Note that we drop those too, later on (see below) */
-        r = drop_privileges(uid, gid,
-                            (UINT64_C(1) << CAP_NET_RAW)|          /* needed for SO_BINDTODEVICE */
-                            (UINT64_C(1) << CAP_NET_BIND_SERVICE)| /* needed to bind on port 53 */
-                            (UINT64_C(1) << CAP_SETPCAP)           /* needed in order to drop the caps later */);
```

Comment:
i'd prefer if that the comments about these three caps aren't lost, they are pretty interesting to read